src/query/fetch.c

```diff
@@ -3871,12 +3872,52 @@ fetch_peek_dbval (THREAD_ENTRY * thread_p, REGU_VARIABLE * regu_var, val_descr *
       REGU_VARIABLE_SET_FLAG (regu_var, REGU_VARIABLE_FETCH_NOT_CONST);
       assert (!REGU_VARIABLE_IS_FLAGED (regu_var, REGU_VARIABLE_FETCH_ALL_CONST));
 
-      /* execute linked query */
-      EXECUTE_REGU_VARIABLE_XASL (thread_p, regu_var, vd);
-      if (CHECK_REGU_VARIABLE_XASL_STATUS (regu_var) != XASL_SUCCESS)
+      if (regu_var->xasl && XASL_IS_FLAGED (regu_var->xasl, XASL_USES_SQ_CACHE)
+	  && !(regu_var->xasl->sq_cache->ht && !regu_var->xasl->sq_cache->enabled))
 	{
-	  goto exit_on_error;
+	  SQ_KEY *key;
+	  if ((key = sq_make_key (thread_p, regu_var->xasl)) == NULL)
```

Comment:
I think the key can be dynamically assigned only when calling sq_put().

Reply:
I applied that on code